packages/inferno/src/DOM/mounting.ts

```diff
@@ -152,7 +159,16 @@ function createClassMountCallback(instance) {
   };
 }
 
-export function mountClassComponentCallbacks(ref, instance, lifecycle: Function[]) {
+function addAppearAnimationHook(animations: AnimationQueues, instance) {
+  // Only do animations in browser
```

Comment:
we can get rid of this "if" statement from runtime by checking const addAppearAnimationHook =  isDocumentAvailable ? emptyFn : function () { ... }

Reply:
Will investigate

Reply:
@Havunen so it appears that the tests will fail if we check for window on loading. Is there some other way you know of that can perform the check later and provide the same optimisation? Is it accessing the window object that is slow? Should we assign a `let addAppearAnimationHook` on first call?

Reply:
infernojs is already doing it: 

https://github.com/infernojs/inferno/blob/bcfcfd8fd786681ab9cb666a7deb39738b34d206/packages/inferno/src/DOM/rendering.ts#L10-L18

Reply:
@Havunen after more investigation and the fact that I couldn't cause an SSR test for animation hooks to fail I realised that the SSR-rendering is done via an entirely different code path that doesn't trigger animation hooks anyway. So I simply remove the if statement. Added test in inferno-server to catch this if SSR-rendering changes in the future.

FIXED in e2c8e0b08